### DISCUSSION TITLE
fixable flaw for HTTP links

### DISCUSSION THREAD

**Discussion by nschonni** [mdn/mdn-community#177] [Platform] [answered]
There are still ~5K HTTP links in the en-us content, and probably a similar amount for the other locales.
I think the HTTP -> HTTPS already happens for the IMG tags, but it would be good for the same thing to happen to `<a>` tags and the inner text.
EX:
1. See `href="http://`
2. Check to see if the HTTPS isn't a 404, otherwise leave it
3. Update `href` and inner text

**Comment by peterbe** [comment]
Yes please! Let's do this. But can we do it slowly?

1. it should be a flaw that a link uses `http://`.
2. make it a fixable flaw where you, like images, just assume that it can be switched to https://
3. always switch it to https:// unconditionally *and* log it as a flaw (if not building a production build)

**Reply by nschonni** [reply]
Is the 1-3 the phases you suggest implementing this as the slow approach?

**Reply by peterbe** [reply]
Yeah, in the sense that that can be 3 separate PRs.

**Comment by hochan222** [comment, marked as answer]
The current discussion is addressed in the discussion at the following link. (resolved)

https://github.com/mdn/mdn-community/discussions/182#discussioncomment-755077
